source/game/field/CourseColMgr.hh

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include "egg/math/BoundBox.hh"
+#include "egg/math/Math.hh"
+#include "egg/math/Vector.hh"
+#include "game/field/RKGndCol.hh"
+
+namespace Field {
+
+typedef bool (
+        RKGndCol::*CollisionCheckFunc)(f32 *distOut, EGG::Vector3f *fnrmOut, u16 *attributeOut);
```

Comment:
Does this `typedef` require to be scoped within `RKGndCol` or could that be omitted?

Reply:
I don't think it requires a scope within `RKGndCol`.

Reply:
To my surprise it does need to be scoped. Otherwise I get `cannot convert 'bool (Field::RKGndCol::*)(float*, EGG::Vector3f*, u16*)' {aka 'bool (Field::RKGndCol::*)(float*, EGG::Vector3f*, short unsigned int*)'} to 'Field::CollisionCheckFunc' {aka 'bool (*)(float*, EGG::Vector3f*, short unsigned int*)'}`